File: src/RequestPath.test.ts
import { strictEqual } from "assert";
import { RequestPath } from "./RequestPath";

describe("Request Path", function () {
  it("gets a path", function () {
        // Arrange
		const url = new URL("https://my-proxy.example.com/path%20to/resource")

		// Act
		const requestPath = new RequestPath(url)
		const result = requestPath.getPath()

		// Assert
		strictEqual(result, "/path to/resource")
  });
});

File: request_path_test.go
package proxy

import (
	"net/url"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestRequestPath(t *testing.T) {

	t.Run("get path", func(t *testing.T) {
		// Arrange
		url, err := url.Parse("https://my-proxy.example.com/path%20to/resource")

		// Act
		requestPath := RequestPath{Url: url}

		result := requestPath.getPath()

		// Assert
		assert.Equal(t, "/path to/resource", result)
		assert.NoError(t, err)
	})
}
